metaflow/plugins/kfp/kfp.py

```diff
@@ -735,7 +747,7 @@ def step_op(
                     preceding_component_inputs=preceding_component_inputs,
                     preceding_component_outputs=preceding_component_outputs,
                 ),
-                base_image=self.base_image,
+                base_image=kfp_component.kfp_step_decorator.attributes["image"] if kfp_component.kfp_step_decorator else self.base_image,
```

Comment:
Linking to the other comment that @talebzeghmi made - if the image is optional you will need to check `kfp_component.kfp_step_decorator.attributes["image"] is not None` even if `kfp_component.kfp_step_decorator` exists